### ISSUE TITLE
[Bug] `Connectivity.ConnectivityChanged` handler does not work when targeting API 34 on Android 14 phones / emulators.

### ISSUE THREAD

**Issue by salvadorjesus** [xamarin/Essentials#2130]
### Description

`Connectivity.ConnectivityChanged` handler does not work when targeting API 34 on Android 14 phones / emulators.

Other combinations of API levels and Android versions seem to work.

This is the **same bug** already *reported and fixed in Maui*: [Adding a Connectivity.ConnectivityChanged handler does not work on Android 14 #19949](https://github.com/dotnet/maui/issues/19949).

### Code snippet

Adding the handler
```
Connectivity.ConnectivityChanged += Connectivity_ConnectivityChanged;
```

Do not call the method:
```
        private async void Connectivity_ConnectivityChanged(object sender, ConnectivityChangedEventArgs e)
        {
            Console.WriteLine("Connectivity changed");
        }
```

### Steps to Reproduce
Download and deploy the [reproduction repo](https://github.com/salvadorjesus/XamarinEssentialsBug-ConnectivityChanged).

Target API 33 or run it on an Android 13 emulator. When you make changes on the connectivity (switch flight mode on and off), the line 'connectivity changed' appears on the console (expected behavior).

When compiling for API 34 *and* running in an Android 14 emulator, nothing appears on console (`Connectivity_ConnectivityChanged` is never reached).

### Reproduction Link

[Reproduction repo](https://github.com/salvadorjesus/XamarinEssentialsBug-ConnectivityChanged).

**Comment by jfversluis** [comment]
Could you try the resulting NuGet from #2132 to verify this fix works?

**Comment by salvadorjesus** [comment]
> Could you try the resulting NuGet from #2132 to verify this fix works?

Has the nugget been generated automatically somewhere with your pull request? I have trouble compiling the nugget locally.

**Comment by jfversluis** [comment]
Yes you should be able to go to [this page](https://dev.azure.com/xamarin/public/_build/results?buildId=109850&view=artifacts&pathAsName=false&type=publishedArtifacts) on Azure DevOps, click on the artifact that says nuget, download it and install it from a local folder.

**Comment by salvadorjesus** [comment]
@jfversluis , it works!

Thank you!
